src/oscar/apps/offer/reports.py

```diff
@@ -42,20 +41,16 @@ class OfferReportGenerator(ReportGenerator):
         'HTML_formatter': OfferReportHTMLFormatter,
     }
 
-    def generate(self):
-        offer_discounts = {}
-        for discount in self.queryset:
-            if discount.offer_id not in offer_discounts:
-                try:
-                    all_offers = ConditionalOffer._default_manager
-                    offer = all_offers.get(id=discount.offer_id)
-                except ConditionalOffer.DoesNotExist:
-                    continue
-                offer_discounts[discount.offer_id] = {
-                    'offer': offer,
-                    'total_discount': D('0.00')
-                }
-            offer_discounts[discount.offer_id]['total_discount'] \
-                += discount.amount
-
-        return self.formatter.generate_response(list(offer_discounts.values()))
+    def get_queryset(self):
+        offers = ConditionalOffer.objects.filter(pk=OuterRef("offer_id"))
+        return super().get_queryset().order_by().values("offer_id").annotate(
+            total_discount=Sum("amount"),
+            offer=Subquery(offers.values("pk")[:1]),
```

Comment:
It would help to have a comment saying this is used in the template, to determine whether to add a link to the offer.